crates/gpui/src/platform/windows/platform.rs

```diff
@@ -83,9 +149,50 @@ impl WindowsPlatform {
             callbacks,
             window_handles,
             event,
+            settings,
         });
         Self { inner }
     }
+
+    /// runs message handlers that should be processed before dispatching to prevent translating unnecessary messages
+    /// returns true if message is handled and should not dispatch
+    fn run_immediate_msg_handlers(&self, msg: &MSG) -> bool {
+        if msg.message == WM_SETTINGCHANGE {
+            self.inner.settings.borrow_mut().update_all();
+            return true;
+        }
+
+        if let Some(inner) = try_get_window_inner(msg.hwnd) {
+            inner.handle_immediate_msg(msg.message, msg.wParam, msg.lParam)
+        } else {
+            false
+        }
+    }
+
+    fn wait_message(&self) -> WindowsMessageWaitResult {
+        let wait_result = unsafe {
+            MsgWaitForMultipleObjects(Some(&[self.inner.event]), false, INFINITE, QS_ALLINPUT)
+        };
+
+        match wait_result {
+            WAIT_EVENT(0) => WindowsMessageWaitResult::ForegroundExecution,
+            WAIT_EVENT(1) => {
+                let mut msg = MSG::default();
+                let get_msg_result = unsafe { GetMessageW(&mut msg, HWND::default(), 0, 0) };
+                // https://learn.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-getmessage
+                if get_msg_result.0 == -1 {
```

Comment:
https://devblogs.microsoft.com/oldnewthing/20130322-00/?p=4873
`GetMessage` can returns -1, but it doesn't fail with valid parameters. We don't have to worry about this, do we?

Reply:
you're right. with the well known params there's no point in checking for `-1`